cpp/CMakeLists.txt

```diff
@@ -14,713 +14,567 @@
 # limitations under the License.
 #=============================================================================
 
-cmake_minimum_required(VERSION 3.14...3.17 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
```

Comment:
Do we want to pin to a specific version's policies to prevent future breakage / unnecessary noise?

Reply:
I think we should keep our policies moving forward with each CMake's release. That will make it easier for us to move our minimum required version as we should encounter less breaking changes.

Reply:
Sounds good, resolving.